src/products/update-product.dto.ts

```diff
@@ -0,0 +1,28 @@
+import {
+  IsBoolean,
+  IsInt,
+  IsNotEmpty,
+  IsOptional,
+  IsString,
+  Max,
+  Min,
+} from 'class-validator';
+import { Transform } from 'class-transformer';
+
+export class UpdateProductDto {
+  @IsOptional()
+  @IsString()
+  @IsNotEmpty()
+  name?: string;
+
+  @IsOptional()
+  @IsInt()
+  @Max(2147483647)
+  @Min(0)
+  priceInPLNgr?: number;
```

Comment:
When it comes to prices we should keep them as string and store them in DB as strings. 

Only if you need to perform calculation you can parse it to Int number and do the calculations and then parse it back to string.

https://stackoverflow.com/questions/3730019/why-not-use-double-or-float-to-represent-currency

P.S. Number in JS is represented in floating point format this is why we cannot 100% trust the calculations. Please get yourselves familiar with the topic